.github/workflows/publish-helminput-plugin-component.yaml

```diff
@@ -0,0 +1,255 @@
+
+      - name: Build multi-arch binaries and generate component constructor
+        working-directory: bindings/go/helm
+        env:
+          VERSION: ${{ steps.calculate_version.outputs.version }}
+        shell: bash
+        run: |
+          set -euo pipefail
+          echo "Generating component constructor with version: $VERSION"
+          task generate/component-constructor
+          test -f "./tmp/component-constructor.yaml" || (echo "component-constructor.yaml not found" && exit 1)
+
+      - name: Attest binaries
+        if: ${{ github.event_name != 'pull_request' }}
+        uses: actions/attest-build-provenance@v3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          subject-path: "bindings/go/helm/tmp/bin/*"
+
+      - name: Upload build artifacts
+        if: ${{ github.event_name != 'pull_request' }}
+        uses: actions/upload-artifact@330a01c490aca151604b8cf639adc76d48f6c5d4 #v5
+        with:
+          name: helm-plugin-build-artifacts
+          path: bindings/go/helm/tmp/
+          retention-days: 1
+
+      - name: Determine if eligible for publishing
+        id: publish-check
+        run: |
+          should_publish="false"
+          # Never publish on pull requests
+          if [[ "${{ github.event_name }}" != "pull_request" ]]; then
+            should_publish="true"
+          fi
+          echo "should_publish=$should_publish" >> "$GITHUB_OUTPUT"
+          echo "✅ Should publish: $should_publish (event: ${{ github.event_name }})"
+
+  publish:
+    name: Publish Component
+    needs: build
+    if: ${{ needs.build.outputs.should_publish == 'true' }}
+    runs-on: ubuntu-latest
+    concurrency:
+      group: helm-plugin-publish-${{ github.ref }}
+      cancel-in-progress: false
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5
+        with:
+          fetch-depth: 0
+          sparse-checkout: |
+            bindings/go/helm
+          sparse-checkout-cone-mode: false
+
+      - name: Download build artifacts
+        uses: actions/download-artifact@018cc2cf5baa6db3ef3c5f8a56943fffe632ef53 #v6
+        with:
+          name: helm-plugin-build-artifacts
+          path: bindings/go/helm/tmp/
+
+      - name: Verify artifacts
+        run: |
+          echo "✅ Verifying build artifacts..."
+          ls -la bindings/go/helm/tmp/
+          test -f bindings/go/helm/tmp/component-constructor.yaml || (echo "❌ component-constructor.yaml missing" && exit 1)
+          test -d bindings/go/helm/tmp/bin || (echo "❌ bin directory missing" && exit 1)
+          echo "✅ All required artifacts present"
+
+      - name: Create Docker config.json for GHCR
+        run: |
+          mkdir -p ${HOME}/.docker
+          echo "{\"auths\": {\"ghcr.io\": {\"auth\": \"$(echo -n "${{ github.actor }}:${{ secrets.GITHUB_TOKEN }}" | base64)\"}}}" > ${HOME}/.docker/config.json
```

Comment:
this needs a cleanup of the token to ensure it is not leaked into the runner at job end